instrumentation/spring/spring-kafka-2.7/library/src/main/java/io/opentelemetry/instrumentation/spring/kafka/v2_7/InstrumentedRecordInterceptor.java

```diff
@@ -31,11 +50,25 @@
     this.decorated = decorated;
   }
 
-  @SuppressWarnings("deprecation") // implementing deprecated method for better compatibility
-  @Override
+  @SuppressWarnings({
+    "deprecation",
+    "unchecked"
+  }) // implementing deprecated method for better compatibility
+  // deprecated method removed in 3.0
+  // @Override
   public ConsumerRecord<K, V> intercept(ConsumerRecord<K, V> record) {
+    if (interceptRecord == null) {
+      return null;
+    }
     start(record);
-    return decorated == null ? record : decorated.intercept(record);
+    if (decorated == null) {
+      return null;
+    }
+    try {
+      return (ConsumerRecord<K, V>) interceptRecord.invoke(decorated, record);
+    } catch (Throwable ignored) {
+      return record;
```

Comment:
Perhaps instead of ignoring we should rethrow the exception as otherwise we could end up swallowing exceptions from the original interceptor that we wrap. Could use the following to rethrow throwable without declaring it
```java
  @SuppressWarnings("unchecked")
  private static <E extends Throwable> void rethrow(Throwable e) throws E {
    throw (E) e;
  }
```

Reply:
Great point, added. Thanks!